### PR TITLE
Fix/e2e test improvements

### LINKED ISSUE #2154: Testnet e2e Tests Failing
With the latest release v1.13.0 live on testnet, the e2e tests are failing: https://github.com/frequency-chain/frequency/actions/runs/10797691465

## Acceptance Criteria

- e2e tests pass

### PR DESCRIPTION
**Pull request by wilwade**
# Goal
The goal of this PR is to add a final set of e2e test improvements to allow e2e tests to run against testnet again post async-backing

Closes #2154 

# Discussion

- Added a way to wait for finalization
- Added errors instead of waiting forever for blocks
- Reduced timeouts thanks to 6s blocktimes on testnet
- Split up more tests for better parallelization

# Testing

- https://github.com/frequency-chain/frequency/actions/runs/11489493946/job/31978401849

### CODE REVIEW

File: e2e/capacity/capacityFail.test.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
+import '@frequency-chain/api-augment';
```
Comment (wilwade) [inline e2e/capacity/capacityFail.test.ts:1]: Moved from `transactions.test.ts`. No changes

File: e2e/capacity/transactions.test.ts [hunk 2]
```diff
@@ -1,10 +1,8 @@
 import '@frequency-chain/api-augment';
```
Comment (wilwade) [inline e2e/capacity/transactions.test.ts:1]: Only moved some tests to `capacityFail.test.ts`. No other changes.

File: e2e/handles/handles.test.ts [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { hasRelayChain } from '../scaffolding/env';
 import { getFundingSource } from '../scaffolding/funding';
 
 const fundingSource = getFundingSource('handles');
-const expirationOffset = hasRelayChain() ? 10 : 100;
+const expirationOffset = hasRelayChain() ? 4 : 100;
```
Comment (wilwade) [inline e2e/handles/handles.test.ts:14]: With testnet at 6s block times, this can be reduced

File: e2e/passkey/passkeyProxy.test.ts [hunk 4]
```diff
@@ -68,7 +68,7 @@ describe('Passkey Pallet Tests', function () {
       const passkeyPayload = await createPasskeyPayload(passKeyPrivateKey, passKeyPublicKey, passkeyCall, false);
       const passkeyProxy = ExtrinsicHelper.executePassKeyProxy(fundedKeys, passkeyPayload);
       assert.doesNotReject(passkeyProxy.fundAndSendUnsigned(fundingSource));
-      await ExtrinsicHelper.runToBlock(await getNextEpochBlock());
+      await ExtrinsicHelper.waitForFinalization();
```
Comment (wilwade) [inline e2e/passkey/passkeyProxy.test.ts:71]: Bugfix. This should have been this all along.

File: e2e/scaffolding/extrinsicHelpers.ts [hunk 5]
```diff
@@ -851,12 +856,38 @@ export class ExtrinsicHelper {
...
   }
 
+  public static async waitForFinalization(ofBlockNumber?: number) {
+    const start = Date.now();
+    const blockNumber = ofBlockNumber || (await getBlockNumber());
+    let currentBlock = await getFinalizedBlockNumber();
+    while (currentBlock < blockNumber) {
+      if (start + 48_000 < Date.now()) {
+        throw new Error(
+          `Waiting for Finalized Block took longer than 48s. Waiting for "${blockNumber.toString()}", Current: "${currentBlock.toString()}"`
```
Comment (wilwade) [inline e2e/scaffolding/extrinsicHelpers.ts:866]: This is a life saver. Some of these were taking forever and it was really hard to track down. Now they are capped.

**Review by JoeCap08055 — APPROVED** [review]

**Pull request merged**